dell/xps/15-9500/nvidia/default.nix

```diff
@@ -0,0 +1,31 @@
+{ lib, pkgs, ... }:
+
+# This creates a new 'nvidia-offload' program that runs the application passed to it on the GPU
+let
+  nvidia-offload = pkgs.writeShellScriptBin "nvidia-offload" ''
+    export __NV_PRIME_RENDER_OFFLOAD=1
+    export __NV_PRIME_RENDER_OFFLOAD_PROVIDER=NVIDIA-G0
+    export __GLX_VENDOR_LIBRARY_NAME=nvidia
+    export __VK_LAYER_NV_optimus=NVIDIA_only
+    exec -a "$0" "$@"
+  '';
+in
+{
+  imports = [
+    ../xps-common.nix
+  ];
+
+  # As per https://nixos.wiki/wiki/Nvidia
+  services.xserver.videoDrivers = lib.mkDefault [ "nvidia" ];
```

Comment:
Can you make a file called: common/gpu/nvidia.nix, where you put the nvidia stuff and include it here?